### ISSUE TITLE
Block Comments Accessibility: reply or comment deleted, focus needs adjusted to relevant comment

### ISSUE THREAD

**Issue by jeffpaul** [WordPress/gutenberg#71678]
### Description

From @t-hamano in https://github.com/WordPress/gutenberg/issues/71249#issue-3333938703:

> Click the delete button in the action dropdown:
>    The screen reader reads `Select an action  menu button  collapsed  subMenu / Add Post ‹ gutenberg — WordPress  document / dialog`. This is redundant. The screen reader should just read `dialog`.

From @joedolson in https://github.com/WordPress/gutenberg/issues/71249#issuecomment-3201284909:

>     I don't get the same announcement; I'm only getting "Select an action menu button collapsed subMenu" as the selection. But I'm slightly unclear at what point you got this announcement. What I notice is that there's a focus loss after deleting a comment, and for me that dropped me in the adminbar; but you may have been dropped somewhere else. Focus loss ends up being delegated to the browser to identify the new location, so that can be highly variable.

> If a reply is deleted, focus should be placed in the parent comment; if a comment is deleted, focus should move to the next comment if there is one, the previous comment if there isn't a next comment, or the parent block if all comments are resolved.

From @adamsilverstein in https://github.com/WordPress/gutenberg/issues/71249#issuecomment-3229794396:

>  If a reply is deleted, focus should be placed in the parent comment

>  If a comment is deleted, focus should move to the next comment if there is one, the previous comment if there isn't a next comment, or the parent block if all comments are resolved

### Step-by-step reproduction instructions

.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by karthick-murugan** [comment]
@jeffpaul - My PR https://github.com/WordPress/gutenberg/pull/71808 covers the third point from @adamsilverstein comments

> If a reply is deleted, focus should be placed in the parent comment

> If a comment is deleted, focus should move to the next comment if there is one, the previous comment if there isn't a next comment, or the parent block if all comments are resolved

**Comment by t-hamano** [comment]
> If a comment is deleted, focus should move to the next comment if there is one, the previous comment if there isn't a next comment, or the parent block if all comments are resolved

This has been fixed in #71974.

> If a reply is deleted, focus should be placed in the parent comment

I can no longer reproduce this either. It may have been fixed as part of another PR.

https://github.com/user-attachments/assets/944a1a44-9a53-4bd9-a917-558ec6c30e52
